### PR TITLE
Improvements to workflow scaling performance testing scripts.

### PR DESCRIPTION
**Pull request by jmchilton**
- Implement a wrapper script for launching a docker-galaxy-stable setup a many ways for this task including:
  - Configure container's Galaxy with outer Galaxy's lib and test directory.
  - Collect log files.
  - Log a statement for interacting with the container easily while the test is running.
  - Disable unneeded services.
- Implement a fan-in-fan-out style workflow for performance testing.
- Add an option to test performance of scheduling workflows without actually running jobs.
- Sample tool for creating large collections easily at beginning of tests (much faster than uploads).
- Sample tool for splitting a file into a collection for fan in and out workflow test.
- Assume Python 2.7 in scripts to simplify things.
- Add documentation.

### CODE REVIEW

**Comment by jmchilton** [comment]
@galaxybot test this.

File: test/functional/tools/for_workflows/create_input_collection.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<tool id="create_input_collection" name="create_input_collection" version="0.1.0">
+  <description>This tool is used to create a collection of text files.</description>
+  <command detect_errors="exit_code">
+    mkdir outputs; cd outputs; python $script
+  </command>
+  <configfiles>
+    <configfile name="script">
+import os
```
Comment (nsoranzo) [inline test/functional/tools/for_workflows/create_input_collection.xml]: `os` is not used.

File: test/functional/tools/for_workflows/create_input_collection.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+for i in range($collection_size):
+    template = "File number %s\n"
+    contents = template % i
+    with open(str(i), "w") as f:
+        f.write(contents)
+</configfile>
```
Comment (nsoranzo) [inline test/functional/tools/for_workflows/create_input_collection.xml:13]: Wrong indentation.

File: test/functional/tools/for_workflows/create_input_collection.xml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  </inputs>
+  <outputs>
+    <collection name="output" type="list" label="lines">
+      <discover_datasets pattern="__name__" directory="outputs" format="txt" />
+    </collection>
+  </outputs>
+  <tests>
+    <test>
+      <param name="collection_size" value="2" />
+      <output_collection name="output" type="list">
```
Comment (nsoranzo) [inline test/functional/tools/for_workflows/create_input_collection.xml]: Add `count="2"`.

File: test/functional/tools/for_workflows/create_input_collection.xml (hunk 2 above)
Comment (jmchilton) [inline test/functional/tools/for_workflows/create_input_collection.xml:13]: I don't want whitespace at the end of my Python script right? What would you suggest here?

Comment (nsoranzo) [inline test/functional/tools/for_workflows/create_input_collection.xml:13]: Isn't Galaxy stripping the content of `configfile`? I think it does for `command`.

Comment (jmchilton) [inline test/functional/tools/for_workflows/create_input_collection.xml:13]: It does strip commands - but it doesn't strip configfiles in tool parsing or evaluation as far as I can see (quickly reviewing the path through the code). I'd see stripping configfiles as bad behavior - it is described as a file so the content should be describing the full file contents.

Comment (nsoranzo) [inline test/functional/tools/for_workflows/create_input_collection.xml:13]: Then I'd do something like https://bitbucket.org/crs4/orione-tools/src/d39acf781fdd5e8ce78382d19520e5c225b81086/custom/exomedepth/exomedepth.xml?at=bb-orione-tools&fileviewer=file-view-default#exomedepth.xml-32 but this is not really that important.

**Comment by jmchilton** [comment]
@galaxybot test this

**Comment by martenson** [comment]
I like the approach very much. Thanks @jmchilton !

**Pull request merged**